src/services/tfjs-service.ts

```diff
@@ -11,14 +11,292 @@ import {Reference, ResourceManager as rmgr} from './resource-manager.js';
 import {logFactory} from '../platform/log-web.js';
 import {Services} from '../runtime/services.js';
 import {requireTf} from '../platform/tf-web.js';
+import {loadImage} from '../platform/image-web.js';
+import {tf} from '../platform/tf-node.js';
 
 const log = logFactory('tfjs-service');
-// Map some TF API to a Service
-const dispose = ({reference}) => rmgr.dispose(reference);
 
-// TODO(alxr) Will add generic ML model service functions in #3094
+type TfTensor = tf.Tensor | tf.Tensor[] | tf.NamedTensorMap;
 
-Services.register('tfjs', {
-  dispose,
+export interface ClassificationPrediction {
+  className: string;
+  probability: number;
+}
+
+abstract class TfModel implements Services {
```

Comment:
Think this type should be `Service` (singular).

Reply:
Good catch